news/management/commands/process_fxa_data.py

```diff
@@ -0,0 +1,154 @@
+            set_fxa_time(fxaid, timestamp)
+
+    print('updated %s fxa timestamps' % update_count)
+    statsd.gauge('process_fxa_data.updates', update_count)
+
+
+def download_fxa_files():
+    s3 = boto3.resource('s3',
+                        aws_access_key_id=settings.FXA_ACCESS_KEY_ID,
+                        aws_secret_access_key=settings.FXA_SECRET_ACCESS_KEY)
+    bucket = s3.Bucket(settings.FXA_S3_BUCKET)
+    for obj in bucket.objects.filter(Prefix=BUCKET_DIR):
+        print('found %s in s3 bucket' % obj.key)
+        tmp_path = TMP.joinpath(obj.key)
+        if not tmp_path.name.endswith('.csv'):
+            continue
+
+        if file_is_done(tmp_path):
+            continue
+
+        if not tmp_path.exists():
+            print('getting ' + obj.key)
+            print('size is %s' % obj.size)
+            tmp_path.parent.mkdir(parents=True, exist_ok=True)
+            try:
+                bucket.download_file(obj.key, str(tmp_path))
+                print('downloaded %s' % tmp_path)
+            except Exception:
+                # something went wrong, delete file
+                print('bad things happened. deleting %s' % tmp_path)
+                tmp_path.unlink()
+
+
+def get_fxa_data():
+    all_fxa_times = {}
+    data_files = DATA_PATH.glob('*.csv')
+    for tmp_path in sorted(data_files):
+        if file_is_done(tmp_path):
+            continue
+
+        print('loading data from %s' % tmp_path)
+        # collect all of the latest timestamps from all files in a dict first
+        # to ensure that we have the minimum data set to compare against SFMC
+        with tmp_path.open() as fxafile:
+            file_count = 0
+            for line in fxafile:
+                file_count += 1
+                fxaid, timestamp = line.strip().split(',')
+                curr_ts = all_fxa_times.get(fxaid, 0)
+                timestamp = int(timestamp)
+                if timestamp > curr_ts:
+                    all_fxa_times[fxaid] = timestamp
+
+            if file_count < 1000000:
+                # if there were fewer than 1M rows we probably got a truncated file
+                # try again later (typically they contain 20M)
+                print('possibly truncated file: %s' % tmp_path)
+            else:
+                set_file_done(tmp_path)
+
+            # done with file either way
+            tmp_path.unlink()
+
+    return all_fxa_times
+
+
+@schedule.scheduled_job('interval', id='process_fxa_data', days=1)
+def main():
+    download_fxa_files()
+    update_fxa_data(get_fxa_data())
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        if not all(getattr(settings, name) for name in ['FXA_ACCESS_KEY_ID',
+                                                        'FXA_SECRET_ACCESS_KEY',
+                                                        'FXA_S3_BUCKET']):
+            raise CommandError('FXA S3 Bucket access not configured')
+
+        main()
+        schedule.start()
```

Comment:
Maybe useful to activate `schedule.start` via an extra flag to the command, something like  `python manage.py process_fxa_data --daemon`. This way we'll be able to run one-offs, specially useful until we get this running?